apps/CMakeLists.txt

```diff
@@ -15,16 +15,39 @@ set(CMAKE_AUTOMOC ON)
 set(CMAKE_AUTORCC ON)
 set(CMAKE_AUTOUIC ON)
 
+list(APPEND CMAKE_AUTOUIC_SEARCH_PATHS "src")
+
+if(VTK_FOUND)
+  set(incs "include/pcl/${SUBSYS_NAME}/render_views_tesselated_sphere.h")
+  set(srcs "src/render_views_tesselated_sphere.cpp")
+endif()
+
+if(QHULL_FOUND)
+  set(incs
+    "include/pcl/${SUBSYS_NAME}/dominant_plane_segmentation.h"
+    "include/pcl/${SUBSYS_NAME}/timer.h"
+    ${incs}
+  )
+  set(impl_incs "include/pcl/${SUBSYS_NAME}/impl/dominant_plane_segmentation.hpp")
+  set(srcs "src/dominant_plane_segmentation.cpp" ${srcs})
+endif()
+
+set(LIB_NAME "pcl_${SUBSYS_NAME}")
+PCL_ADD_LIBRARY(${LIB_NAME} COMPONENT ${SUBSYS_NAME} SOURCES ${srcs} ${impl_incs} ${incs})
+target_link_libraries("${LIB_NAME}" pcl_common pcl_io pcl_filters pcl_visualization pcl_segmentation pcl_surface pcl_features pcl_sample_consensus pcl_search)
+PCL_MAKE_PKGCONFIG(${LIB_NAME} COMPONENT ${SUBSYS_NAME} DESC ${SUBSYS_DESC})
+# Install include files
+PCL_ADD_INCLUDES("${SUBSYS_NAME}" "${SUBSYS_NAME}" ${incs})
+PCL_ADD_INCLUDES("${SUBSYS_NAME}" "${SUBSYS_NAME}/impl" ${impl_incs})
+
 # to be filled with all targets the apps subsystem
 set(PCL_APPS_ALL_TARGETS)
 
-include_directories("${CMAKE_CURRENT_BINARY_DIR}" "${CMAKE_CURRENT_SOURCE_DIR}/include")
-
 PCL_ADD_EXECUTABLE(pcl_test_search_speed COMPONENT ${SUBSYS_NAME} SOURCES src/test_search.cpp)
 target_link_libraries(pcl_test_search_speed pcl_common pcl_io pcl_search pcl_kdtree pcl_visualization)
 
 PCL_ADD_EXECUTABLE(pcl_nn_classification_example COMPONENT ${SUBSYS_NAME} SOURCES src/nn_classification_example.cpp)
-target_link_libraries(pcl_nn_classification_example pcl_common pcl_io pcl_features pcl_kdtree)
+target_link_libraries(pcl_nn_classification_example pcl_apps pcl_common pcl_io pcl_features pcl_kdtree)
```

Comment:
Because it includes pcl/apps/vfh_nn_classifier.h - which resides in apps "library".

Reply:
Line 21 the old include_directories made this available to all the executables afterwards.

Reply:
But `vfh_nn_classifier.h` seems to be a standalone header with every function fully defined, so no corresponding cpp file that would make linking to pcl_apps necessary. Wouldn't it be better to add `target_include_directories(pcl_nn_classification_example ...)` instead of unnecessarily adding `pcl_apps` as a dependency?

Reply:
Yeah, that should do it aswell of course. I'll change asap.